.github/workflows/release.yml

```diff
@@ -0,0 +1,33 @@
+name: release
+on: workflow_dispatch
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Verify branch name
+        if: github.ref != 'refs/heads/main'
+        run: |
+          echo "🚨 The release must start from the main branch!"
+          exit 1
+
+      - name: Configure releaser details
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+      - name: Install pnpm
+        uses: wyvox/action-setup-pnpm@v3
+        with:
+          node-version: 18.x
+          node-registry-url: "https://registry.npmjs.org"
```

Comment:
I think you will also need `pnpm-version` as it is not pinned with `packageManager` in the package.json

Reply:
Ah yes, good point! I will add `pnpm-version: 8.6.0`